### ISSUE TITLE
Traffic Problem with class forager

### ISSUE THREAD

**Issue by betrox007** [Courseplay/courseplay#5347]
hello Dear all, First of all thanks for the efforts for still developing this mod,here i have some problem
when using Class DLC for unload combine, the tractor always stuck at traffic while follow combine from behind,i found this this problem at the map i playing on, so i decided to try at vanilla map, load all mod i use in my map,and the problem is same,when using class forager and class axion tractor, they always stuck at traffic, when the tractor behind combine, i don't know may be this is the Giant problem for the collision box, or may be CP need to adjust a little bit the front and end node for class equipment,but if using big chrone forager, this traffic only warning for a few second.

here i attached my log, and some screenshoot,hopely can help

for error 187 ignore it, i was forget to detach trailler and reattach before change assign combine,

the error 187 happen when change combine, from class jaguar to big chrone and  do not detach trailer from tractor,
this error disappear when i detach trailer,then reattach, select grain transport mode, then select unload combine mode, then assign combine to tractor.
for error 640 i don't know what is that..

thanks and best regards

[log.txt](https://github.com/Courseplay/courseplay/files/4602784/log.txt)

![fsScreen_2020_05_09_12_09_44](https://user-images.githubusercontent.com/56646850/81465566-1b582d00-91f5-11ea-949b-ea9fd5a445e7.png)
![fsScreen_2020_05_09_12_32_35](https://user-images.githubusercontent.com/56646850/81465705-1f387f00-91f6-11ea-9d6e-e3efdac57c3e.png)
![fsScreen_2020_05_09_12_32_47](https://user-images.githubusercontent.com/56646850/81465714-2d869b00-91f6-11ea-9283-b4428954e424.png)

![fsScreen_2020_05_09_11_47_39](https://user-images.githubusercontent.com/56646850/81465581-32971a80-91f5-11ea-8ca8-43b6b6cc1a8c.png)
![fsScreen_2020_05_09_12_11_00](https://user-images.githubusercontent.com/56646850/81465582-3460de00-91f5-11ea-940f-d7cb24c4bd80.png)
![fsScreen_2020_05_09_11_48_56](https://user-images.githubusercontent.com/56646850/81465595-52c6d980-91f5-11ea-9c6e-ed875f672ae9.png)
![fsScreen_2020_05_09_12_21_59](https://user-images.githubusercontent.com/56646850/81465596-54909d00-91f5-11ea-8559-e8bfcd81ff0b.png)
![fsScreen_2020_05_09_12_22_20](https://user-images.githubusercontent.com/56646850/81465605-61ad8c00-91f5-11ea-81de-f0478c265aef.png)
![fsScreen_2020_05_09_11_50_47](https://user-images.githubusercontent.com/56646850/81465606-63774f80-91f5-11ea-8cc1-c2064b2ee130.png)
![fsScreen_2020_05_09_12_26_57](https://user-images.githubusercontent.com/56646850/81465612-6a9e5d80-91f5-11ea-9748-2d99f430a143.png)
![fsScreen_2020_05_09_12_31_14](https://user-images.githubusercontent.com/56646850/81465700-092abe80-91f6-11ea-86b1-26a7c313ae12.png)

**Comment by Tensuko** [comment]
Somehow only the big X can throw far enough that it works the best. We will find a solution for this later.
Then, right now it can cause problem when you have more then one unloader assigned to a combine, like they meet each other on a long way to combine.
But those Errors are kinda 'intetesting' as I never tried this. Will try it out and check on this.

**Comment by betrox007** [comment]
the error 187, because i want to try the differences between big x and jaguar, so at the begining i use jaguar, all tractor assign to jaguar on combine selection, when change combine to big x, i'm just stop all tractor then directly select big x at combine selector and sold the jaguar, then that error happen and big x can't call them to approach, so i stop them again, and detach trailer from tractor, then reattach and select mode 1, after that select mode 2 and then sellect big x at combine selector, everything back to normal.
and yea i think for now the solution i will only using big x, as this forager have a long trow

thanks for reply

**Comment by SteelRodent** [comment]
FWIW: The tracked Claas Jaguar has some odd issues that are entirely on Giants; as is noticeable when Vanilla drivers use an incredible amount of space to turn it around and struggle to line it up correctly on lane changes. Also on a sidenote: This machine is 130 hp short of what's needed to use the 9 m headers at full speed. Don't know if CP is able to ignore this, but you need 750+ hp to use the 9 m headers, which I believe is also why the pipe on the tracked Jaguar is so short. The wheeled Jaguar is however available with Extra-long pipe, which is longer than the Krone, so I don't see why it wouldn't work with CP (haven't tested it yet,  but without CP is has longer range than the Krone).

**Comment by Tensuko** [comment]
That is not the issue we have but good to know..

**Comment by 0Pietro20** [comment]
As far as I have been able to prove, the problem lies not only in the length of the tube, but above all where the rear marker is positioned (when you activate the debug channel 4 you see it behind the forage harvester). If this is so far back that the tractors stop too early (it regulates the advance of the tractor and prevents it from crashing on the forage harvester) then even with the extra long tube (I personally tried with the John Deere forage harvester) it cannot discharge properly. I hope that in the future this problem will be solved (it would be better if the CP did it automatically but also with the manual setting I would be satisfied). I would like to remind (I opened an issue on purpose) that during the side unloading the tractor tends to go too far ahead of the forage harvester
